pkg/manifests/config.go

```diff
@@ -456,11 +456,12 @@ func (c *Config) Precheck() error {
 
 	// Highlight deprecated config fields.
 	var d float64
-	if c.ClusterMonitoringConfiguration.K8sPrometheusAdapter.DedicatedServiceMonitors != nil {
+	if c.ClusterMonitoringConfiguration.K8sPrometheusAdapter != nil {
```

Comment:
I suspect that `c.ClusterMonitoringConfiguration.K8sPrometheusAdapter` is never nil because of https://github.com/openshift/cluster-monitoring-operator/blob/701e6d3e44bfb42f911a56bc0831ac1e670509ad/pkg/manifests/config.go#L279-L281

Reply:
right, I moved this to deployment as Adapter Deployment method won't be called.